### PR TITLE
Replace gopkg.in/yaml.v2 with sigs.k8s.io/yaml

### LINKED ISSUE #1456: Remove gopkg.in/yaml.v2
/kind cleanup
/area provider/ibmcloud

**What steps did you take and what happened:**
gopkg.in/yaml.v2 repository is almost stale, lets replace with the internal k8s yaml module - sigs.k8s.io/yaml

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by prashantrewar**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1456 

**Special notes for your reviewer**:

/area provider/ibmcloud
/kind cleanup

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mkumatag** [comment]
@prashantrewar please fix the CLA

**Comment by prashantrewar** [comment]
@mkumatag done!

File: cmd/capibmadm/cmd/version/version.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"gopkg.in/yaml.v2"
+	"sigs.k8s.io/yaml"
```
Comment (mkumatag) [inline cmd/capibmadm/cmd/version/version.go:26]: please move `sigs.k8s.io/yaml` after line number 28 in the section of `sigs.k8s.io/` imports

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cmd/capibmadm/cmd/version/version.go (hunk 1 above)
Comment (prashantrewar) [reply cmd/capibmadm/cmd/version/version.go]: Done. Could you please share how you figured out the change based on the error message. I tried running `make lint` and `make lint-fix` but it didn't help much.

Comment (mkumatag) [reply cmd/capibmadm/cmd/version/version.go]: Here is the failure message:
```
cmd/capibmadm/cmd/version/version.go:27: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/IBM) -s prefix(k8s.io) -s prefix(sigs.k8s.io) -s prefix(sigs.k8s.io/cluster-api) -s prefix(sigs.k8s.io/cluster-api-provider-ibmcloud) -s blank -s dot (gci)
	"sigs.k8s.io/cluster-api-provider-ibmcloud/version"
cmd/capibmadm/cmd/version/version.go:28: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(github.com/IBM) -s prefix(k8s.io) -s prefix(sigs.k8s.io) -s prefix(sigs.k8s.io/cluster-api) -s prefix(sigs.k8s.io/cluster-api-provider-ibmcloud) -s blank -s dot (gci)
	"sigs.k8s.io/yaml"
```

one of the lint failing wrt to [gci](https://golangci-lint.run/usage/linters/#gci), and the rule set for this project is https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/61e7587fadf00e0ece903645732e9c2b55323a41/.golangci.yml#L59C1-L69C12

So the import order should in this case is:

```golang
import (
	"encoding/json"
	"fmt"
	"io"

	"github.com/pkg/errors"
	"github.com/spf13/cobra"
	"gopkg.in/yaml.v2"

	"sigs.k8s.io/yaml"

	"sigs.k8s.io/cluster-api-provider-ibmcloud/version"
)
```

Comment (prashantrewar) [reply cmd/capibmadm/cmd/version/version.go:27]: Thank you so much for the explanation @mkumatag! I have made the changes and `make lint ` is passing now.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**